Lesson6.ipynb

```diff
@@ -27,7 +34,686 @@
+    "        self.name = name\n",
+    "        self.age = age\n",
+    "        self.teamName = teamName\n",
+    "        self.isCaptain = \"Yes\" if isCaptain else \"No\"       #assign \"Yes\" if the isCaptian value is True else assigns \"No\"\n",
+    "        self.totalStepCount = totalStepCount\n",
+    "        Member.totalMembers += 1                            #totalMembers keeps increases by 1, everytime a new Member object is created\n",
+    "\n",
+    "    def printDetails(self):\n",
+    "        print(\"Member:\", self.name, \" Team:\", self.teamName, \" Total Steps:\", self.totalStepCount, \"Captain:\", getattr(self,\"isCaptain\"), \"\\n\")\n",
+    "                                                            #getattr method here returns the value of isCaptain. it's an alias for self.isCaptain \n",
+    "john = Member(\"John\", \"18\", \"Safire\", True, 500000)         #Member object creation\n",
+    "john.printDetails()\n",
+    "print(Member.__doc__, \"\\n\")\n",
+    "print(Member.totalMembers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Inheritance\n",
+    "\n",
+    "In object-oriented programming, inheritance enables new objects to take on the properties of existing objects. A class that is used as the basis for inheritance is called **a superclass** or **base class**. A class that inherits from a superclass is called **a subclass** or **derived class**. The terms **parent class** and **child class** are also acceptable terms to use respectively. A child inherits visible properties and methods from its parent while adding additional properties and methods of its own.\n",
+    "\n",
+    "Derived classes are declared much like their parent class; however, a list of base classes to inherit from is given after the class name in parathesis. Below is the syntax\n",
+    "\n",
+    "```python\n",
+    "   class SubClassName (ParentClass1[, ParentClass2, ...]):\n",
+    "       'Optional class documentation string'\n",
+    "       class_suite\n",
+    "```\n",
+    "> **super(SubClassName, Instance) ** return a proxy object that delegates method calls to a parent or sibling class of type. This is useful for accessing inherited methods that have been overridden in a class. \n",
+    "\n",
+    ">The **issubclass(sub, sup)** boolean function returns true if the given subclass sub is indeed a subclass of the superclass sup.\n",
+    "\n",
+    ">The **isinstance(obj, Class)** boolean function returns true if obj is an instance of class Class or is an instance of a subclass of Class\n",
+    "\n",
+    "\n",
+    "*\n",
+    "**Overriding** is a feature that enables a child class to provide different implementation for a method that is already defined and/or implemented in its parent class or one of its parent classes. The overriden method in the child class should have the same name, signature, and parameters as the one in its parent class.*\n",
```

Comment:
"Overriding is a feature that enables a child class to provide different implementation for a method that is already defined and/or implemented in its parent class or one of its parent classes. The overriden method in the child class should have the same name, signature, and parameters as the one in its parent class."
Should be read
"Overriding is a feature that enables a child class to provide a different implementation for a method that is already defined and/or implemented in its parent class or grand-parent class. The overridden method in the child class must have the same name, signature, and parameters as the one in its parent class in order to be overridden."